specification/src/specification/queries/filtering.md

```diff
@@ -93,24 +70,11 @@ Binary comparison operators are denoted by expressions with a `type` field of `b
 
 See type [`ComparisonValue`](../../reference/types.md#comparisonvalue) for the valid inhabitants of the `value` field.
 
+_Note_: in general, `PathElement`s in a `ComparisonValue` can refer to _array_ relationships. However, in the case of the `in` operator, such requests can be very difficult to implement, and for practical purposes, it is not very useful to express such queries. Therefore, connectors can expect `PathElements` in `ComparisonValue`s to always only refer to _object_ relationships, and fail with a `Bad Request` error otherwise.
```

Comment:
> Therefore, connectors can expect `PathElements` in `ComparisonValue`s to always only refer to _object_ relationships

Is this always true, or only true for the `in` operator? If always true, this would allow simplifying SQL a great deal, paths become simple left joins.

However, there's a drawback to not supporting this* at all, at least in clickhouse. In terms of usefulness, it's effectively an exists check with a comparison. However, the combination of exists + root columns comparisons cannot be implemented soundly without correlated subqueries, while paths that effectively evaluate to arrays of values are much easier to implement.

*where this is paths that traverse array relationships. I have no issues with removing right hand array relationships in paths for binary array comparison operators